jobs/build/ocp/Jenkinsfile

```diff
@@ -674,16 +674,25 @@ images:build
               catch ( err ) {
                 record_log = buildlib.parse_record_log( OIT_WORKING )
                 builds = record_log['build']
-                failed_map = [:]
+                failed_build_map = [:]
+                failed_push_list = []
                 for(i = 0; i < builds.size(); i++) {
                   bld = builds[i]
                   distgit = bld['distgit']
                   if ( bld['status'] == '-1' ){
-                    failed_map[distgit] = bld['task_url']
+                    failed_build_map[distgit] = bld['task_url']
                   }
                   else {
                     // build may have succeeded later. If so, remove.
-                    failed_map.remove(distgit)
+                    failed_build_map.remove(distgit)
+                  }
+                  
+                  if ( bld['push_status'] == '-1' ){
+                    failed_push_list += distgit
+                  }
+                  else {
+                    // push may have succeeded later. If so, remove.
+                    failed_push_list.removeAll{ it == distgit }
```

Comment:
I would just put everything in the same failed_map, otherwise your build exclusion logic breaks down. The key:value can be `<dist-git-name>:"failed to push..."` . For simplicity, we don't need to discriminate between build & push failures in the email.

Reply:
What I was going for was an image push was basically non-fatal. But that's fine. Will simplify.